contracts/modules/governance/MemberRoles.sol

```diff
@@ -280,43 +280,55 @@ contract MemberRoles is Governed, Iupgradable {
   }
 
   /**
-   * @dev Called by existed member if wish to switch membership to other address.
-   * @param _add address of user to forward membership.
+   * @dev switches membership for msg.sender to the specified address.
+   * @param newAddress address of user to forward membership.
    */
-  function switchMembership(address _add) external {
+  function switchMembership(address newAddress) external {
+    _switchMembership(msg.sender, newAddress);
+    tk.transferFrom(msg.sender, newAddress, tk.balanceOf(msg.sender));
```

Comment:
Move this to `_switchMembership` and change `msg.sender` to `member`

Reply:
as discussed this is done to allow Cover.sol to receive the NXM approval instead of MemberRoles to hide the implementation details. should stay here for that to work